src/main/java/com/artipie/metrics/Metrics.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.metrics;
+
+/**
+ * Registry of metrics by name.
+ *
+ * @since 0.6
+ */
+public interface Metrics {
```

Comment:
@olegmoz it seems to me that using separate interfaces `Counter` and `Gauge` will lower cohesion because those two interfaces will operate on the same set of encapsulated data. I propose two alternatives:
Only one interface:
```java
interface Metrics {
     void inc();
     void add(int num);
     void set(long val);
     long value(); 
}
```  
or split metrics to two different interfaces `CounterMetrics` `GaugeMetrics`. wdyt?

Reply:
@Vatavuk separating counters and gauges are very common in monitoring systems, see DataDog, Prometheus etc. The nature of data is very different: one is for collecting incrementing changes from multiple sources, another is mostly for getting states from single source (think requests count vs current number of requests being handled). Implementation will probably differ even for in memory implementation (gauge could be implemented with just a `volitile` value, when counter will require `AtomicLong` or `synchronized` sections). Splitting `Metrics` interfaces does not sound to me as a great idea eaither, because we will need to pass all of them to adapters. It's easier to do with single object for all metrics.

Reply:
@olegmoz ok, thx for the clarification